qt/python/mantidqt/widgets/sliceviewer/peaksviewer/representation/ellipsoid.py

```diff
@@ -102,6 +105,17 @@ def draw(cls, peak_origin, peak_shape, slice_info, painter, fg_color, bg_color):
         return Painted(painter, artists)
 
 
+def convert_spherical_representation_to_ellipsoid(shape_info):
+    # convert shape_info dict from sphere to ellipsoid for plotting
+    for key in ['radius', 'background_inner_radius', 'background_outer_radius']:
+        shape_info[f"{key}{0}"] = shape_info.pop(key)
```

Comment:
If you set PeakRadius=BackgroundOuterRadius in the IntegratePeaksMD call the background_outer_radius property of the sphere isn't present (it isn't set in the PeakShapeSpherical.cpp code). And you get an error from sliceviewer from this line:
![image](https://user-images.githubusercontent.com/56295817/117035639-75061100-acfc-11eb-920c-60f7000167f6.png)

Not entirely sure what this combination of sphere properties is supposed to mean. The original spherical.py unit that you've deleted seemed to interpret it as meaning "no background". So perhaps the code here should work if the background_inner_radius and background_outer_radius properties are missing?

Reply:
Ah yes, thanks for spotting this, there are various ways to not set the background to be integrated and it can produce different shape definitions - I missed this one!